.github/workflows/ci.yaml

```diff
@@ -0,0 +1,59 @@
+name: CI
+
+on: push
+jobs:
+  test_and_lint:
+    strategy:
+      fail-fast: false
+      matrix:
+        ci_node_total: [8]
+        ci_node_index: [0, 1, 2, 3, 4, 5, 6, 7]
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:11-alpine
+        ports:
+          - '5432:5432'
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: password
+          POSTGRES_DB: dfe-teachers-payment-service_test
+    env:
+      RAILS_ENV: test
+      DFE_TEACHERS_PAYMENT_SERVICE_DATABASE_USERNAME: postgres
+      DFE_TEACHERS_PAYMENT_SERVICE_DATABASE_PASSWORD: password
+      DFE_TEACHERS_PAYMENT_SERVICE_DATABASE_HOST: localhost
+      CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
+      CI_NODE_INDEX: ${{ matrix.ci_node_index }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+      - name: Setup Node
+        uses: actions/setup-node@v4
+      - name: Setup yarn
+        run: npm install -g yarn
+      - name: Install Node packages
+        run: yarn install
+      - name: Install Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: 3.2.0
+          bundler-cache: true
+      - name: Precompile assets
+        run: bundle exec rails assets:precompile
```

Comment:
We probably want to cache this. Also the `test:prepare` task is probably better to use so we don't have to manage other steps if other dependencies are introduced. Here's a snippet I wrote for another project:

```
    - name: Restore assets cache
      id: assets-cache
      uses: actions/cache/restore@v4
      with:
        path: |
          public/assets
        key: assets-${{ hashFiles('**/app/assets/**/*', '**/app/javascript/**/*') }}
        restore-keys: |
          ${{ runner.os }}-assets-

    - name: Prepare assets
      if: ${{ steps.assets-cache.outputs.cache-hit != 'true' }}
      run: bin/rails test:prepare

    - name: Save assets cache
      if: ${{ steps.assets-cache.outputs.cache-hit != 'true' }}
      uses: actions/cache/save@v4
      with:
        path: |
          public/assets
          public/packs-test
        key: ${{ steps.assets-cache.outputs.cache-primary-key }}
```

Reply:
We don't have `test:prepare` task, do you think it's better to add a task that calls `assets:precompile` or just leave the call to `assets:precomplie` in action?

Reply:
If it’s not there then no; I can’t recall how this is defined and it’s probably not worth worrying about.